posthog/temporal/data_imports/sources/stripe/source.py

```diff
@@ -0,0 +1,93 @@
+from typing import cast
+from posthog.schema import ExternalDataSourceType, SourceConfig, SourceFieldInputConfig, Type4
+from posthog.temporal.data_imports.sources.common.base import BaseSource, FieldType
+from posthog.temporal.data_imports.sources.common.registry import SourceRegistry
+from posthog.temporal.data_imports.sources.common.schema import SourceSchema
+
+from posthog.temporal.data_imports.sources.stripe.stripe import (
+    StripePermissionError,
+    stripe_source,
+    validate_credentials as validate_stripe_credentials,
+)
+from posthog.temporal.data_imports.sources.stripe.settings import (
+    ENDPOINTS as STRIPE_ENDPOINTS,
+    INCREMENTAL_FIELDS as STRIPE_INCREMENTAL_FIELDS,
+)
+from posthog.temporal.data_imports.pipelines.pipeline.typings import SourceInputs, SourceResponse
+from posthog.temporal.data_imports.sources.generated_configs import StripeSourceConfig
+from posthog.warehouse.models import ExternalDataSource
+
+
+@SourceRegistry.register
+class StripeSource(BaseSource[StripeSourceConfig]):
+    @property
+    def source_type(self) -> ExternalDataSource.Type:
+        return ExternalDataSource.Type.STRIPE
+
+    @property
+    def get_source_config(self) -> SourceConfig:
+        return SourceConfig(
+            name=ExternalDataSourceType.STRIPE,
+            caption="""Enter your Stripe credentials to automatically pull your Stripe data into the PostHog Data warehouse.
```

Comment:
I wanna add a small counterpoint to this approach: implementing this via markdown (or backend-controlled strings) severely limits what we can do with sources setup. I'm likely to work on improving onboarding/setup for Stripe and this will restrict what I can do.

Are you against this kind of configuration living in the client side at all? We can probably do backend-controlled by default but some sources might require more complex frontend work. Is that reasonable?